src/nvim/autocmd.c

```diff
@@ -954,6 +955,7 @@ static int do_autocmd_event(event_T event,
       ac->cmd = vim_strsave(cmd);
       ac->script_ctx = current_sctx;
       ac->script_ctx.sc_lnum += sourcing_lnum;
+      nlua_set_sctx(&ac->script_ctx);
```

Comment:
optional/future: this block of code setting various fields on `script_ctx` looks repeated in many places throughout the codebase. Could be valuable to encapsulate it in a fuction.

Reply:
Good idea . Maybe in a future pr .